src/qibosoq/programs/flux.py

```diff
@@ -94,8 +94,11 @@ def execute_flux_pulse(self, pulse: Pulse):
             i_vals = np.full(duration, amp)
         elif isinstance(pulse, FluxExponential):
             i_vals = pulse.i_values(duration, max_gain)
+        elif isinstance(pulse, Arbitrary):
+            i_vals = np.array(pulse.i_values)
+            logger.info("Arbitrary shaped flux pulse. q_vals will be ignored.")
         else:
-            raise NotImplementedError("Only Rectangular and FluxExponential flux pulses are supported")
+            raise NotImplementedError("Only Rectangular, FluxExponential and Arbitrary are supported for flux pulses")
```

Comment:
I would have to add another "switch" that could be a bit complex (since for every pulse I should get I-values).
This feature would also probably never be used...